### PR TITLE
fix: DetrayMaterialConversion for Gen2

### PR DESCRIPTION
**Pull request by asalzburger**
The Detray material conversion was broken due to missing locality of the index map.

It renames the cache object to simply `Cache` since it does more now, and through the namespace it gets clear that it is the conversion cache.

### CODE REVIEW

File: Plugins/Detray/include/Acts/Plugins/Detray/DetrayConversionUtils.hpp [hunk 1]
```diff
@@ -28,12 +32,37 @@ namespace DetrayConversionUtils {
...
+      : detectorVolumes(dVolumes) {}
+
+  /// The volumes of the detector for index lookup
+  std::vector<const Acts::Experimental::DetectorVolume*> detectorVolumes;
   /// This is a map to pass on volume link information
   std::map<GeometryIdentifier, unsigned long> volumeLinks;
+  /// This is a multimap to pass volume local surface link information
+  /// The portal splitting requires a multimap implementation here
+  ///
+  /// These are voljme local, heence indexed per volumes
```
Comment (niermann999) [inline Plugins/Detray/include/Acts/Plugins/Detray/DetrayConversionUtils.hpp:49]:
```suggestion
  /// These are volume local, hence indexed per volumes
```

File: Plugins/Detray/src/DetrayMaterialConverter.cpp [hunk 2]
```diff
@@ -80,18 +80,26 @@ Acts::DetrayMaterialConverter::convertHomogeneousSurfaceMaterial(
...
+            auto surfaceIndices =
+                localSurfaceLinks->second.equal_range(surface->geometryId());
+            // Loop over the equal range and fill one grid each, this is needed
+            // as the initial portal could be split into multiple surfaces
+            for (auto itr = surfaceIndices.first; itr != surfaceIndices.second;
+                 ++itr) {
+              // Make an identified link copy for every matching surface
+              detray::io::single_link_payload surfaceLink;
+              surfaceLink.link = itr->second;
+              slabPayload.surface = surfaceLink;
```
Comment (niermann999) [inline Plugins/Detray/src/DetrayMaterialConverter.cpp:97]:
```suggestion
              slabPayload.surface.link = itr->second;
```

File: Plugins/Detray/src/DetrayMaterialConverter.cpp [hunk 3]
```diff
@@ -248,39 +256,49 @@ Acts::DetrayMaterialConverter::convertGridSurfaceMaterial(
...
+              localSurfaceLinks->second.equal_range(surface->geometryId());
+
+          ACTS_VERBOSE(
+              "DetrayMaterialConverter: assigning to "
+              << std::distance(surfaceIndices.first, surfaceIndices.second)
+              << " surfaces with material in volume " << volume->name());
+          DetrayMaterialGrid materialGrid =
+              convertGridSurfaceMaterial(*surface->surfaceMaterial(), logger);
+          // Ignore if an empty payload is returned
+          if (materialGrid.axes.empty() && materialGrid.bins.empty()) {
```
Comment (niermann999) [inline Plugins/Detray/src/DetrayMaterialConverter.cpp:284]: Should this not be an "or" actually?
```suggestion
          if (materialGrid.axes.empty() || materialGrid.bins.empty()) {
```

**Review by niermann999 — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [3ea63a021225bc0909fd3d692f7cefdf33469918]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887080)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887079)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887078)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887077)||
|:yellow_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887076)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887075)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887074)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887073)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887072)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887071)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887070)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887069)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887068)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887067)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887066)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887065)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887064)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887063)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887062)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887061)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887060)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887059)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887058)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887057)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887056)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887055)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887054)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887053)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887052)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887051)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887050)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887049)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44887048)||

**Pull request merged**